mod_reforged/hooks/skills/special/double_grip.nut

```diff
@@ -297,6 +293,11 @@
 				}
 			}
 		}
+
+		if (_skill.getID() != "actives.puncture" && this.m.CurrWeaponType == ::Const.Items.WeaponType.Dagger)
+		{
+			_properties.MeleeDamageMult *= 1.2;
```

Comment:
I prefer if you do the vanilla approach but all within this double_grip.nut file.
- So first you grant the damage bonus during `onUpdate` so that it is displayed in the characters stats page how a player would 
- Then during `onAnySkillUsed` you divide the `MeleeDamageMult` through the same value so that it is hopefully back to the normal damage value for puncture only

For that you should then introduce a shared member variable for the dagger damage mult